examples/synthetics_example_test.go

```diff
@@ -135,133 +147,400 @@ func pickPrivateAgentID(ctx context.Context) (string, error) {
+			Target: "www.example.com",
+		},
+	})
+
+	test.Settings.Period = 15 * time.Minute
+	test.Settings.Family = synthetics.IPFamilyV4
+	test.Settings.NotificationChannels = []string{} // must contain IDs of existing notification channels
+	test.Settings.Health = synthetics.HealthSettings{
+		LatencyCritical:           50 * time.Millisecond,
+		LatencyWarning:            20 * time.Millisecond,
+		LatencyCriticalStdDev:     100 * time.Millisecond,
+		LatencyWarningStdDev:      100 * time.Millisecond,
+		JitterCritical:            50 * time.Millisecond,
+		JitterWarning:             20 * time.Millisecond,
+		JitterCriticalStdDev:      100 * time.Millisecond,
+		JitterWarningStdDev:       100 * time.Millisecond,
+		PacketLossCritical:        100,
+		PacketLossWarning:         100,
+		HTTPLatencyCritical:       50 * time.Millisecond,
+		HTTPLatencyWarning:        20 * time.Millisecond,
+		HTTPLatencyCriticalStdDev: 100 * time.Millisecond,
+		HTTPLatencyWarningStdDev:  100 * time.Millisecond,
+		HTTPValidCodes:            []uint32{http.StatusOK, http.StatusCreated},
+		DNSValidCodes:             []uint32{1, 2, 3},
+		UnhealthySubtestThreshold: 2,
+		AlarmActivation: &synthetics.AlarmActivationSettings{
+			TimeWindow:  75 * time.Minute,
+			Times:       4,
+			GracePeriod: 3,
+		},
+	}
+	test.Settings.Ping.Delay = 100 * time.Millisecond
+	test.Settings.Ping.Port = 65535
+	test.Settings.Traceroute.Port = 1
+
+	return test
+}
+
+func createMinimalTests() error {
+	ctx := context.Background()
+	client, err := NewClient()
+	if err != nil {
+		return err
+	}
+
+	for _, t := range []*synthetics.Test{
+		newMinimalIPTest(),
+		newMinimalNetworkGridTest(),
+		newMinimalHostnameTest(),
+		newMinimalAgentTest(),
+		newMinimalNetworkMeshTest(),
+		newMinimalFlowTest(),
+		newMinimalURLTest(),
+		newMinimalPageLoadTest(),
+		newMinimalDNSTest(),
+		newMinimalDNSGridTest(),
+	} {
+		err = createAndDeleteTest(ctx, client, t)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createAndDeleteTest(ctx context.Context, client *kentikapi.Client, test *synthetics.Test) error {
+	fmt.Println("### Creating synthetic test", test.Name)
+	test, err := client.Synthetics.CreateTest(ctx, test)
+	if err != nil {
+		return fmt.Errorf("client.Synthetics.CreateTest: %w", err)
+	}
+
+	fmt.Println("Created synthetic test:")
+	PrettyPrint(test)
+
+	fmt.Println("### Deleting synthetic test", test.Name)
+	err = client.Synthetics.DeleteTest(ctx, test.ID)
+	if err != nil {
+		return fmt.Errorf("client.Synthetics.DeleteTest: %w", err)
+	}
+	fmt.Printf("Deleted synthetic test %q successfully\n", test.Name)
+	return nil
+}
+
+func newMinimalIPTest() *synthetics.Test {
+	return synthetics.NewIPTest(synthetics.IPTestRequiredFields{
+		BasePingTraceTestRequiredFields: synthetics.BasePingTraceTestRequiredFields{
+			BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+				Name:     "go-sdk-example-minimal-ip-test",
+				AgentIDs: []string{"2592"},
```

Comment:
If this is always executed against a "fake" server, then hardcoding agent IDs is fine. Otherwise, agent IDs need to be selected by matching available agent by some criteria.